component/main.jsonnet

```
@@ -5,6 +5,144 @@ local inv = kap.inventory();
+      'app.kubernetes.io/name': params.namespace,
+      'topolvm.cybozu.com/webhook': 'ignore',
+    },
+  },
+};
+
+local lvmd_configmap = kube.ConfigMap('topolvm-lvmd0') {
+  metadata+: {
+    labels+: {
+      idx: '0',
+      'app.kubernetes.io/name': 'topolvm',
+      'app.kubernetes.io/instance': 'topolvm',
+      'app.kubernetes.io/version': '0.8.3',
+    },
+  },
+  data: {
+    'lvmd.yaml': |||
+      socket-name: /run/topolvm/lvmd.sock
+      device-classes:
+      - name: %(name)s
+        default: %(default)s
+        spare-gb: %(sparegb)d
+        volume-group: %(volumegroup)s
+    ||| % {
+      name: params.deviceclasses[0].name,
+      default: 'true',
+      sparegb: params.deviceclasses[0].sparegb,
+      volumegroup: params.deviceclasses[0].volumegroup,
+    },
+  },
+};
+
+local lvmd_daemonset = kube.DaemonSet('topolvm-lvmd0') {
+  metadata+: {
+    labels+: {
+      idx: '0',
+      'app.kubernetes.io/name': 'topolvm',
+      'app.kubernetes.io/instance': 'topolvm',
+      'app.kubernetes.io/version': params.helmValues.image.tag,
+    },
+  },
+  spec: {
+    selector: {
+      matchLabels: {
+        idx: '0',
+        'app.kubernetes.io/name': 'topolvm-lvmd0',
+      },
+    },
+    template: {
+      metadata: {
+        labels: {
+          idx: '0',
+          'app.kubernetes.io/name': 'topolvm-lvmd0',
+        },
+        annotations: {
+          'prometheus.io/port': '8080',
+        },
+      },
+      spec: {
+        serviceAccountName: 'topolvm-lvmd',
+        hostPID: true,
+        containers: [
+          {
+            name: 'lvmd',
+            image: params.helmValues.image.repository + ':' + params.helmValues.image.tag,
+            securityContext: {
+              privileged: true,
+            },
+            command: [
+              '/lvmd',
+              '--container',
+            ],
+            resources: {
+              limits: {
+                cpu: params.resources.lvmd.limits.cpu,
+                memory: params.resources.lvmd.limits.memory,
+              },
+              requests: {
+                cpu: params.resources.lvmd.requests.cpu,
+                memory: params.resources.lvmd.requests.memory,
+              },
+            },
+            volumeMounts: [
+              {
+                name: 'lvmd-socket-dir',
+                mountPath: '/run/topolvm',
+              },
+              {
+                name: 'config',
+                mountPath: '/etc/topolvm',
+              },
+            ],
+          },
+        ],
+        volumes: [
+          {
+            name: 'config',
+            configMap: {
+              name: 'topolvm-lvmd0',
+            },
+          },
+          {
+            name: 'lvmd-socket-dir',
+            hostPath: {
+              path: '/run/topolvm',
+              type: 'DirectoryOrCreate',
+            },
+          },
+        ],
+        nodeSelector: {
+          'syn.tools/topolvm': '',
+        },
+      },
+    },
+  },
+};
+
```

Here you'll have to also generate the storage classes with something like (not sure I managed to copy all relevant bits from the StorageClass definition in the Helm chart):

```suggestion

// Uses component `storageclass`, default class annotation is handled by the component lib of `storageclass`.
local sc = import 'lib/storageclass.libsonnet';
local make_sc_config(sc_params) =
  com.makeMergeable({
    allowVolumeExpansion: params.allowVolumeExpansion,
    parameters: {
      'csi.storage.k8s.io/fstype': params.fstype,
      'topolvm.cybozu.com/device-class': params.device_class
    }
  });
local storageclasses = [
  sc.StorageClass(name) {
    provisioner: 'topolvm.cybozu.com',
    volumeBindingMode: 'WaitForFirstConsumer',
  } + make_sc_config(params.storageclasses[name]
  for name std.objectFields(params.storageclasses)
];
```